.travis.yml

```diff
@@ -8,26 +8,33 @@ php:
   - 7.0
   - hhvm
 
-matrix:
-  allow_failures:
-    - php: 7.0
-    - php: hhvm
+sudo: false
 
 cache:
   directories:
     - $HOME/.composer/cache
+    - vendor
+
+matrix:
+  include:
+    - php: 5.4
+      env: COMPOSER_FLAGS="--prefer-lowest"
+  allow_failures:
+    - php: 7.0
+    - php: hhvm
 
 before_script:
-  - composer self-update
-  - composer install --prefer-source --no-interaction
+  - mkdir -p ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d && echo "memory_limit=-1" >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
+  - composer selfupdate
+  - composer config -q github-oauth.github.com $GITHUB_OAUTH_TOKEN
+  - travis_wait composer update --prefer-dist --no-interaction $COMPOSER_FLAGS
```

Comment:
No. Using `composer install` will avoid all the interest of testing a library.

Installed as a library via composer, users won't be synced to your lock file.

Libraries must be tested every time with both lowest and latest dependencies to cover all cases.

composer.lock should not be here. It's only useful for project, not library. But I think you had scrutinizer issue without it...

BTW, `composer install` will fail with `--prefer-lowest` option.

All library projects I see (except this one) use `composer update` for Travis test.

Regards.